swagger.yaml

```diff
@@ -3715,36 +3794,51 @@ definitions:
             feature_id:
               type: string
               description: The ID of the feature
-              enum:
-              - 'ADDITIONAL_NTP_SOURCE'
-              - 'REQUESTED_HOSTNAME'
-              - 'PROXY'
-              - 'SNO'
-              - 'DAY2_HOSTS'
-              - 'VIP_AUTO_ALLOC'
-              - 'DISK_SELECTION'
-              - 'OVN_NETWORK_TYPE'
-              - 'SDN_NETWORK_TYPE'
-              - 'PLATFORM_SELECTION'
-              - 'SCHEDULABLE_MASTERS'
-              - 'AUTO_ASSIGN_ROLE'
-              - 'CUSTOM_MANIFEST'
-              - 'DISK_ENCRYPTION'
-              - 'CLUSTER_MANAGED_NETWORKING_WITH_VMS'
-              - 'ARM64_ARCHITECTURE'
-              - 'ARM64_ARCHITECTURE_WITH_CLUSTER_MANAGED_NETWORKING'
-              - 'PPC64LE_ARCHITECTURE'
-              - 'S390X_ARCHITECTURE'
-              - 'SINGLE_NODE_EXPANSION'
-              - 'LVM'
-              - 'DUAL_STACK_NETWORKING'
-              - 'MULTIARCH_RELEASE_IMAGE'
-              - 'NUTANIX_INTEGRATION'
-              - 'DUAL_STACK_VIPS'
-              - 'USER_MANAGED_NETWORKING_WITH_MULTI_NODE'
+              $ref: '#/definitions/feature-support-level-id'
             support_level:
-              type: string
-              enum: [supported, unsupported, tech-preview, dev-preview]
+              $ref: '#/definitions/support-level'
+
+  support-levels:
+    type: object
+    description: Map of featureID alongside their support level
+    additionalProperties:
+      $ref : '#/definitions/support-level'
+
+  support-level:
+    type: string
+    enum: [ supported, unsupported, tech-preview, dev-preview ]
+
+  feature-support-level-id:
+    type: string
+    enum:
+      - 'ADDITIONAL_NTP_SOURCE'
+      - 'REQUESTED_HOSTNAME'
+      - 'PROXY'
+      - 'SNO'
+      - 'DAY2_HOSTS'
+      - 'VIP_AUTO_ALLOC'
+      - 'DISK_SELECTION'
+      - 'OVN_NETWORK_TYPE'
+      - 'SDN_NETWORK_TYPE'
+      - 'PLATFORM_SELECTION'
+      - 'SCHEDULABLE_MASTERS'
+      - 'AUTO_ASSIGN_ROLE'
+      - 'CUSTOM_MANIFEST'
+      - 'DISK_ENCRYPTION'
+      - 'CLUSTER_MANAGED_NETWORKING_WITH_VMS'
+      - 'X86_64_ARCHITECTURE'
+      - 'ARM64_ARCHITECTURE'
+      - 'PPC64LE_ARCHITECTURE'
+      - 'S390X_ARCHITECTURE'
```

Comment:
These should be in a separate enum.  It looks like it makes the diff smaller, but is it the best way?

Reply:
Moving those will not break the API? Also, they are still representing a features, just from a "different kind"